src/v/datalake/translation/partition_translator.cc

```diff
@@ -391,16 +402,17 @@ ss::future<> partition_translator::close() noexcept {
 }
 
 scheduling::translation_status partition_translator::status() const {
-    // Biggest TODO: Currently we do not track intervals of target lag.
-    // So the scheduler is pretty much scheduling on defaults
-    // Track intervals return the correct remaining duration / memory
-    // reservations. The latter is already exposed as mux::buffered_bytes()
-    return {};
+    return scheduling::translation_status{
+      .target_lag = _lag_tracking->target_lag(),
+      .next_checkpoint_deadline = _lag_tracking->next_checkpoint_deadline(),
+      .memory_bytes_reserved = _translation_ctx->buffered_bytes(),
+      .translation_backlog = _lag_tracking->translation_backlog(),
+    };
 }
 
 ss::future<std::optional<std::chrono::milliseconds>>
 partition_translator::current_lag_ms() const {
-    return _data_source->current_lag_ms(wait_timeout);
+    co_return _lag_tracking->current_lag_ms();
```

Comment:
nitpick: any reason this still needs to return a future?

Reply:
no, will change this